src/EnergyPlus/DataHeatBalance.cc

```diff
@@ -374,8 +374,8 @@ void CheckAndSetConstructionProperties(EnergyPlusData &state,
 
     thisConstruct.DayltPropPtr = 0;
     InsideMaterNum = thisConstruct.LayerPoint(InsideLayer);
-    auto const *thisMaterialInside = state.dataMaterial->Material(InsideMaterNum);
     if (InsideMaterNum != 0) {
+        auto const *thisMaterialInside = dynamic_cast<const Material::MaterialChild *>(state.dataMaterial->Material(InsideMaterNum));
```

Comment:
I think it may be good to add `AbsorpThermal`, `AbsorpSolar`, and `AbsorpVisible` to `MaterialBase` because they are used quite often. I would leave the EMS and `VarAbsorp` stuff in `MaterialChild` though.

Reply:
Maybe AbsorpThermalFront and AbsorpThermalBack should also be moved to MaterialBase with AbsorpThermal?